StripeFinancialConnections/StripeFinancialConnections/Source/Native/Shared/DataAccessNoticeViewController.swift

```diff
@@ -12,13 +12,16 @@ import UIKit
 final class DataAccessNoticeViewController: SheetViewController {
 
     private let dataAccessNotice: FinancialConnectionsDataAccessNotice
+    private let theme: FinancialConnectionsTheme?
     private let didSelectUrl: (URL) -> Void
 
     init(
         dataAccessNotice: FinancialConnectionsDataAccessNotice,
+        theme: FinancialConnectionsTheme?,
```

Comment:
what are scenarios where theme is NULL? does it make sense that it should be NULL? or should we always have a default value?

What I'm thinking of is that doesn't that extra NULL just add extra complexity in handling another state? 

But all depends on whether there's a need for a NULL

Reply:
Main reason the theme is optional is because it's an optional property in the [backend manifest model](https://livegrep.corp.stripe.com/view/stripe-internal/pay-server/lib/bank_connections/api/resource/link_account_session_manifest_resource.rb#L59), so we have it optional in the client-side manifest model. 

Another reason, which hasn't surfaced anywhere in this PR, is that there's situations where having a null theme is "valid". For instance, when the flow is doing its initial load to fetch the manifest, we show a spinner view, and that view should be themed. Since we don't have a theme from the manifest yet, we'll use a null theme, and color the view neutrally rather than using Stripe or Link colors.

That being said, we _could_ have the client-side model have a non-nullable `theme` property, and fallback on a "default" theme when we either haven't loaded it yet, or the value sent back from the server is null. Though, since we don't have a guarantee the server will give us a theme in the synchronize response, I lean towards keeping it optional. Let me know what you think!

Reply:
edit: I actually didn't realize how you avoided scattering nullability logic throughout the code:

```
extension FinancialConnectionsTheme? {
```

its an extension on the optional...

but then what happens if you don't use it as an optional? it seems like its a compiler error? if true that seems off?

---

as discussed in the 1:1, feel free to do what's best as you are in the best position to make the judgement call

the key thing to think about is whether there's ever a valid reason for auth flow UI code (consent etc.) to ever NOT have a theme (I'm excluding the rare exceptions before synchronize); IF it's true that we always want to have a theme in certain parts, then we should likely not have an optional there

Reply:
>but then what happens if you don't use it as an optional

That's a good point I hadn't thought of. Will look into this all!

Reply:
I'm a little on the fence about what the best approach here is. With the field being optional in the manifest server response, we need an optional theme field on the client side model. i.e. 

```swift
enum FinancialConnectionsTheme: String, SafeEnumCodable, Equatable {
    case light = "light"
    case dashboardLight = "dashboard_light"
    case linkLight = "link_light"
    case unparsable
}

let theme: Theme?
```

What we _could_ do, is have an abstraction to this model, which is non-nullable, and absorbs the null-case into the default `light` theme. Something like this: 

```swift
enum FinancialConnectionsAppearance {
    case light
    case linkLight
    
    init(from theme: FinancialConnectionsTheme?) {
        switch theme {
        case .linkLight:
            self = .linkLight
        case .light, .dashboardLight, .unparsable, .none:
            self = .light
        }
    }
}
```

Reason for naming it `Appearance` is because the `theme` namespace is already reserved by the synthesized decodable model (unless we want to specify custom `CodingKeys` 😅). Then, the manifest model would have to look something like this: 

```swift
private let theme: FinancialConnectionsTheme?
var appearance: FinancialConnectionsAppearance {
    Appearance(from: theme)
}
```

With this change, we solve the issue of passing around a nullable theme instance, but we add complexity to the client side manifest model and possible confusion by having both a `theme` and `appearance` field. To help with this, we can keep `theme` private to indicate `appearance` is the field to use (but it does suck losing the "theme" namespace).

@kgaidis-stripe do you have a preference between what I proposed above, or keeping it all as is? Or maybe even an alternative approach? I'm fine either way, but if we chose to go with the above suggestion, I think that, despite the additional noise, using custom `CodingKeys` would be great so that we can continue to use the "theme" namespace, and access it via `manifest.theme`.

Reply:
Take a look at the [latest commit](https://github.com/stripe/stripe-ios/pull/3793/commits/25a388380bbfd0412bd3c48b7cf0f07e9c68e251) for my suggested approach!